### PR TITLE
Adding initial code stub with tests

### PR DESCRIPTION
**Pull request by hsorby**
resolves issue #2

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+set(_PARAM_ANNOTATION "Enable memcheck testing.")
+if(CMAKE_COMPILER_IS_GNUCXX)
+  set( LIBCELLML_MEMCHECK OFF CACHE BOOL ${_PARAM_ANNOTATION} )
+endif()
+if( MEMCHECK AND CMAKE_COMPILER_IS_GNUCXX )
+  set( LIBCELLML_MEMCHECK "${MEMCHECK}" CACHE BOOL ${_PARAM_ANNOTATION} FORCE )
+endif()
+
+# UNIT_TESTS ==> LIBCELLML_UNIT_TESTS
+set(_PARAM_ANNOTATION "Enable build of automated LibCellML tests.")
```
Comment (nickerso) [inline CMakeLists.txt]: LibCellML ==> libCellML

File: tests/test_template.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+# Set the test name, 'test_' will be prepended to the
+# name set here
+set(CURRENT_TEST <test-name>)
+# Set a category name to enable running commands like:
+#    ctest -R <category-label>
+# which will run the tests matching this category-label.
+# Can be left empty (or just not set)
+set(CURRENT_CATEGORY <category-label>)
+list(APPEND YAWOT_TESTS ${CURRENT_TEST})
```
Comment (nickerso) [inline tests/test_template.cmake]: should maybe be APPEND LIBCELLML_TESTS ?

**Comment by nickerso** [comment]
seems good apart from a couple of minor comments inline. And the buildbot isn't working?

**Comment by nickerso** [comment]
Great, that all looks awesome to me.

**Comment by agarny** [comment]
This looks good to me, so closing.

**Pull request merged**